src/core/hrs.ml

```diff
@@ -42,30 +42,35 @@ let print_term : bool -> term pp = fun lhs ->
         out "pi(%a,\\v_%s.%a)" pp a (Bindlib.name_of x) pp b
   in pp
 
-(** [print_rule oc s r] outputs the rule declaration corresponding [r] (on the
-    symbol [s]), to the output channel [oc]. *)
-let print_rule : Format.formatter -> sym -> rule -> unit = fun oc s r ->
-  (* Print the pattern variable declarations. *)
-  let get_patt_names : term list -> string list = fun ts ->
+(** [print_rule oc lhs rhs] outputs the rule declaration [lhs]->[rhs]
+    to the output channel [oc] *)
+let print_rule : Format.formatter -> term -> term -> unit = fun oc lhs rhs ->
+  (* gets pattern and binded variable names *)
+  let get_var_names : term -> StrSet.t = fun t ->
     let names = Stdlib.ref [] in
     let fn t =
       match t with
-      | Patt(_,n,_) -> Stdlib.(names := n :: !names)
+      | Patt(_,n,_) -> Stdlib.(names := ("&" ^ n) :: !names)
+      | Abst(_,b) ->
+        let (x, _) = Bindlib.unbind b in
+        let n = Bindlib.name_of x in
+        Stdlib.(names := ("v_" ^ n) :: !names)
```

Comment:
This prefix could generate a name that may clash with pattern variable names. Why do you add a prefix?